classes/local/step/trigger_event.php

```diff
@@ -0,0 +1,281 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_dataflows\local\step;
+
+use tool_dataflows\local\event_processor;
+use tool_dataflows\local\execution\iterators\iterator;
+use tool_dataflows\local\execution\iterators\dataflow_iterator;
+use tool_dataflows\local\execution\flow_engine_step;
+use tool_dataflows\local\execution\engine_step;
+use tool_dataflows\local\execution\engine;
+/**
+ * Event reader step
+ *
+ * @package   tool_dataflows
+ * @author    Matthew Hilton <matthewhilton@catalyst-au.net>
+ * @copyright 2022, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class trigger_event extends trigger_step {
+    /**
+     * Executes the reader event by consuming the event from the queue and add the event data to the variables for this step.
+     *
+     * @param mixed|null $input
+     * @return mixed
+     */
+    public function execute($input = null) {
+        if (!$this->enginestep->engine->isdryrun) {
+            event_processor::consume_event($input->id);
+        }
+
+        $variables = $this->get_variables();
+        $variables->set('event', $input->eventdata);
+
+        return true;
+    }
+
+    /**
+     * A list of outputs and their description if applicable.
+     *
+     * @return array of outputs
+     */
+    public function define_outputs(): array {
+        return ['event' => get_string('trigger_event:variable:event', 'tool_dataflows')];
+    }
+
+    /**
+     * Returns the event iterator. Reads events from the database queue table and returns them as an array iterator.
+     *
+     * @return iterator Array iterator containing the events queued for this dataflow.
+     */
+    public function get_iterator(): iterator {
+        $dataflowid = $this->stepdef->get('dataflowid');
+        $eventarray = event_processor::get_events_for_dataflow($dataflowid);
+
+        // Re-parse the stringified JSON event data for each.
+        $eventarray = array_map(function ($e) {
+            $e->eventdata = json_decode($e->eventdata);
+            return $e;
+        }, $eventarray);
+
+        return new dataflow_iterator($this->enginestep, new \ArrayIterator($eventarray));
+    }
+
+    /**
+     * Get the default data.
+     *
+     * @param \stdClass $data
+     * @return stdClass
+     */
```

Comment:
```suggestion
     * @param \stdClass $data
     * @return \stdClass
     */
```
Types should be consistently declared.